src/include/gurt/atomic.h

```diff
@@ -43,7 +43,18 @@
 #if HAVE_STDATOMIC
 
 #include <stdatomic.h>
+
+
+#if defined(__cplusplus)
+extern "C" {
+#endif
+
+#ifdef __INTEL_COMPILER
+#define ATOMIC volatile
```

Comment:
This looks good, it would be great to have the atomic tests extended so they use the currently failing function, although given you change to SConscript above are the tests actually running icc?

Reply:
the preprocessor was not inherited from the top level sconstruct, so for the clang test the code is preprocessed by icc then compiled by clang.